Source/Applications/SystemCenter/Model/Security/SecurityGroup.cs

```diff
@@ -278,7 +283,31 @@ protected override DataTable GetSearchResults(PostData postData, int? page)
                 row["DisplayName"] = group.DisplayName;
             }
 
-            // #ToDo Add Filtering by DisplayName and Type
+            DataTable filteredDataTable = dataTable.Clone();
+            foreach (Search search in postData.Searches)
+            {
+                IEnumerable<DataRow> filteredRows = dataTable.AsEnumerable();
+
+                switch(search.Operator)
+                {
+                    case "=":
+                        filteredRows = filteredRows.Where((row) => row.Field<string>(search.FieldName).ToLower() == search.SearchText.ToLower());
+                        break;
+                    case "LIKE":
+                        filteredRows = filteredRows.AsEnumerable().Where((row) => row.Field<string>(search.FieldName).ToLower().Contains(search.SearchText.Trim('*').ToLower()));
+                        break;
+                    case "NOT LIKE":
+                        filteredRows = filteredRows.AsEnumerable().Where((row) => !row.Field<string>(search.FieldName).ToLower().Contains(search.SearchText.Trim('*').ToLower()));
+                        break;
+                }
+
+                foreach (DataRow row in filteredRows)
+                {
+                    filteredDataTable.Rows.Add(row.ItemArray);
+                }
+
+                dataTable = filteredDataTable;
+            }
```

Comment:
```suggestion
            }
            foreach (DataRow row in filteredRows)
            {
                filteredDataTable.Rows.Add(row.ItemArray);
            }

            dataTable = filteredDataTable;
```
Further filters does not filter down further, the way it's setup now is like a logical AND rather than a logical OR